main.go

```diff
@@ -64,57 +67,68 @@ func main() {
-				Body: &closeComment,
-			})
+	go func() {
+	commentLoop:
+		for {
+			comments, _, err := client.Issues.ListComments(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueListCommentsOptions{})
 			if err != nil {
-				fmt.Printf("error commenting on issue: %v\n", err)
+				fmt.Printf("error getting comments: %v\n", err)
 				os.Exit(1)
 			}
-			_, _, err = client.Issues.Edit(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueRequest{State: &newState})
-			if err != nil {
-				fmt.Printf("error closing issue: %v\n", err)
-				os.Exit(1)
-			}
-			break commentLoop
-		case approvalStatusDenied:
-			newState := "closed"
-			closeComment := "Request denied. Closing issue and failing workflow."
-			_, _, err := client.Issues.CreateComment(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueComment{
-				Body: &closeComment,
-			})
+
+			approved, err := approvalFromComments(comments, approvers, minimumApprovals)
 			if err != nil {
-				fmt.Printf("error commenting on issue: %v\n", err)
+				fmt.Printf("error getting approval from comments: %v\n", err)
 				os.Exit(1)
 			}
-			_, _, err = client.Issues.Edit(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueRequest{State: &newState})
-			if err != nil {
-				fmt.Printf("error closing issue: %v\n", err)
+			fmt.Printf("Workflow status: %s\n", approved)
+			switch approved {
+			case approvalStatusApproved:
+				newState := "closed"
+				closeComment := "All approvers have approved, continuing workflow and closing this issue."
+				_, _, err := client.Issues.CreateComment(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueComment{
+					Body: &closeComment,
+				})
+				if err != nil {
+					fmt.Printf("error commenting on issue: %v\n", err)
+					os.Exit(1)
+				}
+				_, _, err = client.Issues.Edit(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueRequest{State: &newState})
+				if err != nil {
+					fmt.Printf("error closing issue: %v\n", err)
+					os.Exit(1)
+				}
+				break commentLoop
+			case approvalStatusDenied:
+				newState := "closed"
+				closeComment := "Request denied. Closing issue and failing workflow."
+				_, _, err := client.Issues.CreateComment(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueComment{
+					Body: &closeComment,
+				})
+				if err != nil {
+					fmt.Printf("error commenting on issue: %v\n", err)
+					os.Exit(1)
+				}
+				_, _, err = client.Issues.Edit(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueRequest{State: &newState})
+				if err != nil {
+					fmt.Printf("error closing issue: %v\n", err)
+					os.Exit(1)
+				}
 				os.Exit(1)
 			}
-			os.Exit(1)
+
+			time.Sleep(pollingInterval)
 		}
 
-		time.Sleep(pollingInterval)
+		fmt.Println("Workflow manual approval completed")
+		os.Exit(0)
```

Comment:
Exit the goroutine cleanly if the loop completes

Reply:
Can you rework this approach so that we exit the process from the main goroutine instead of this one? I think that approach will be cleaner and more readable.

Reply:
Thanks for the feedback!

I've moved initialising the signal handler into its own function, with a nested goroutine that handles closing the issue and leaving a comment. This means there's no big changes to the main function, other than a call to the new function prior to `commentLoop:`. 

I think this is a lot cleaner, the diff is easier to read now.